web/client/actions/tutorial.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright 2017, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const {setControlProperty} = require('./controls');
+const assign = require('object-assign');
+const START_TUTORIAL = 'START_TUTORIAL';
+const CLOSE_TUTORIAL = 'CLOSE_TUTORIAL';
+const REMOVE_INTRO_TUTORIAL = 'REMOVE_INTRO_TUTORIAL';
+const SET_STEPS_TUTORIAL = 'SET_STEPS_TUTORIAL';
+const TOGGLE_TUTORIAL = 'TOGGLE_TUTORIAL';
+const CHANGE_STATUS_TUTORIAL = 'CHANGE_STATUS_TUTORIAL';
+
+function startTutorial() {
+    return {
+        type: START_TUTORIAL
+    };
+}
+
+function closeTutorial() {
+    return {
+        type: CLOSE_TUTORIAL
+    };
+}
+
+function removeIntroTutorial(intro) {
+    return {
+        type: REMOVE_INTRO_TUTORIAL,
+        intro: intro
+    };
+}
+
+function setStepsTutorial(steps) {
+    return {
+        type: SET_STEPS_TUTORIAL,
+        steps: steps
+    };
+}
+
+function toggleTutorial() {
+    return {
+        type: TOGGLE_TUTORIAL
+    };
+}
+
+function changeStatusTutorial(status) {
+    return {
+        type: CHANGE_STATUS_TUTORIAL,
+        status: status
+    };
+}
+
+function setupTutorial(introSteps) {
+    return (dispatch, getState) => {
```

Comment:
Using getState in actions is usually evil: reducers are the place where we manipulate state